### DISCUSSION TITLE
Compatibility Issue with mageworx and the Latest Version of swissup/module-breeze

### DISCUSSION THREAD

**Discussion by marctbb** [breezefront/community#235] [3. Integrations]
Hey there,

Just wanted to drop a quick note about an issue I've run into after updating the "swissup/module-breeze" module to its latest version 2.12. It seems that with this update, the "swissup/breeze" dependency is added at version "^2.0," causing conflicts with the customizable options from Mageworx.

It appears that the Mageworx extension isn't fully compatible with the latest "swissup/module-breeze" version, resulting in errors in my Magento 2 store. The quick workaround was to either disable the Breeze Experience in the store admin under Content -> Configuration -> Edit Theme -> Breeze -> Enable Breeze Experience (set to No), or by deactivating the Mageworx option.

I've attached a screenshot of the console error and the Mageworx configuration for reference. If anyone else has encountered this or has insights into a solution, it would be awesome to hear your thoughts.

Cheers!

![error_custom_option](https://github.com/breezefront/community/assets/138032384/65677294-3356-48f0-8808-6f2441cc4325)
![mageworx_config](https://github.com/breezefront/community/assets/138032384/0d19a109-20db-49f3-ba65-bf513ad43be6)

**Comment by vovayatsyuk** [comment]
Can you please check if this commit fixes it? https://github.com/breezefront/module-breeze/commit/c7cf06924e4bbfdc61daa226d2e8c5f98347f46d

**Reply by marctbb** [reply]
Hi vovayatsyuk thanks for your reply. It still doesn't work with the patch you indicated. Thanks

**Reply by vovayatsyuk** [reply]
I can't reproduce it locally. Do you have a URL for your site?

**Reply by marctbb** [reply]
Hi, this is an environment url: https://test.getstepr.co.uk/stepr.html

**Reply by vovayatsyuk** [reply]
Thanks. 

The patch is not applied on the server. Make sure you've applied it and redeploy static content.

**Reply by marctbb** [reply]
Can you see if the patch is applied now? Thanks

**Reply by vovayatsyuk** [reply]
The patch is applied now, but it didn't help because it requires some additional patches for component.js file.

For now, I recommend reverting the patch I gave and applying the following patch instead:

```diff
diff --git a/view/frontend/web/js/core/main.js b/view/frontend/web/js/core/main.js
index e892cb7..b1a6392 100644
--- a/view/frontend/web/js/core/main.js
+++ b/view/frontend/web/js/core/main.js
@@ -336,7 +336,7 @@
 
             if (_.isFunction(component)) {
                 instance = component(data.settings, el);
-                if (!instance || !instance.component) {
+                if (!component._proto && !instance?.component) {
                     instance = component;
                 }
             } else if (_.isObject(component) && _.isFunction(component[name])) {
@@ -345,8 +345,10 @@
                 return;
             }
 
-            $(el).component(name, instance);
-            $.registry.set(name, el, instance);
+            if (instance) {
+                $(el).component(name, instance);
+                $.registry.set(name, el, instance);
+            }
         });
     });
 
```

**Reply by marctbb** [reply]
Hello, it works, thank you very much. The console error has disappeared. On the other hand, do you know why these errors occur? They already appeared before the update. I would appreciate it very much on your part.
Regards
![input_error](https://github.com/breezefront/community/assets/138032384/381bbf60-70d9-48b6-b100-20fbdae4bbc0)

**Reply by vovayatsyuk** [reply]
You need to apply the following patches (we've updated them recently): https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions?tab=readme-ov-file#required-patches

**Reply by marctbb** [reply]
Hi I have updated the patches and added the last one I was missing mageworx/module-optionadvancedpricing/view/base/web/js/advanced-pricing.js . In principle I have applied them well, but the errors still appear in the console. What can I do?
Thanks

**Reply by vovayatsyuk** [reply]
You can check if the bundle is using updated files. Maybe you need to redeploy static content. 

p.s. I recommend disabling bundling during development to get updated files without redeployment.

**Reply by marctbb** [reply]
I applied it locally and it did not work when I applied the patch. Neither did deploying in the test environment. Locally I have tested the patch applied in breeze theme and it still doesn't work.
Thanks

**Reply by vovayatsyuk** [reply]
You can use search by code to see where the error comes from:

https://github.com/breezefront/community/assets/306080/3767d20a-d1bb-42b4-b2ea-0d87b4b5859c

Looks like you should patch this code?

**Comment by marctbb** [comment]
Hello, yes sorry, the patch had not been applied properly. Now no longer appears any error by console with that url. My code shares Mageworx with another website in which when I apply all the patches you indicated, I get a similar error, the difference is that this website does not use breeze. But when I apply the patch to work on breeze, it fails on the others. Attached screenshot and url
https://test.forceusa.co.uk/g12tm-all-in-one-trainer-functional-trainer-90-5-kg-smith-machine-power-rack-and-leg-press.html
![error](https://github.com/breezefront/community/assets/138032384/4e576bb3-522c-4703-acf9-3e0df16fb218)
![patchApplied](https://github.com/breezefront/community/assets/138032384/d35dd907-0cbf-4930-8417-bf80ad80729c)
[patches.tar.gz](https://github.com/breezefront/community/files/14125291/patches.tar.gz)

Thanks

**Reply by vovayatsyuk** [reply]
My version of the Mageworx module doesn't have this code so I can't improve integration instructions. Can you please contact [swissuplabs](https://swissuplabs.com/contacts/) and provide the source code for this module?

**Reply by vovayatsyuk** [reply]
Just realized that you can modify these lines as follows to be compatible with jQuery and Cash:

```js
options.filter($.breeze ? 'input:not([type="hidden"])' : 'input[type!="hidden"]')
```
